bndtools.core/src/bndtools/editor/BndEditor.java

```diff
@@ -700,6 +703,8 @@ public void dispose() {
                 .removeResourceChangeListener(this);
         }
 
+        LaunchUtils.endRun((Run) model.getProject());
```

Comment:
Is this safe? `model` still holds a reference to the run which is now _ended_.

Reply:
it was like that before so I'm not sure how much less safe it is that before.

Reply:
Well the difference. is that you are now telling RunListeners about something that is edit mode. That is new.